hoomd/md/ForceComposite.cc

```diff
@@ -973,6 +986,42 @@ void ForceComposite::updateCompositeParticles(uint64_t timestep)
         updated_pos += dr_space;
         quat<Scalar> updated_orientation = orientation * local_orientation;
 
+        // calculate the tangential velocity of constituents
+        const Scalar4 vel_mass = h_velocity.data[particle_index];
+        double mass = vel_mass.w;
+        quat<Scalar> angvel_body = conj(orientation) * angmom * 0.5;
+        vec3<Scalar> av_body = angvel_body.v;
+        if (I.x != Scalar(0))
+            {
+            av_body.x /= I.x;
+            }
+        else
+            {
+            av_body.x = Scalar(0);
+            }
+
+        if (I.y != Scalar(0))
+            {
+            av_body.y /= I.y;
+            }
+        else
+            {
+            av_body.y = Scalar(0);
+            }
+
+        if (I.z != Scalar(0))
+            {
+            av_body.z /= I.z;
+            }
+        else
+            {
+            av_body.z = Scalar(0);
+            }
+
+        vec3<Scalar> tangential_velocity = cross(av_body, local_pos);
+        // update velocity of constituents
+        vec3<Scalar> updated_vel = rotate(orientation, tangential_velocity) + vel;
```

Comment:
This comment belongs above before this code block. Adding comments about the math being done would also be helpful here.
```suggestion
        vec3<Scalar> updated_vel = vel + rotate(orientation, cross(av_body, local_pos));
```